samples/net/mqtt/src/modules/network/network_lte.c

```diff
@@ -72,44 +50,44 @@ void pdn_event_handler(uint8_t cid, enum pdn_event event, int reason)
 	}
 }
 
+static void connectivity_event_handler(struct net_mgmt_event_callback *cb,
+				       uint32_t event,
+				       struct net_if *iface)
+{
+	if (event == NET_EVENT_CONN_IF_FATAL_ERROR) {
+		LOG_ERR("Fatal error received from the connectivity layer");
+		SEND_FATAL_ERROR();
+		return;
+	}
+}
+
 static void network_task(void)
 {
 	int err;
+	struct net_if *net_if = net_if_get_default();
 
-	/* Initialize the modem library. */
-	err = nrf_modem_lib_init();
-	if (err) {
-		LOG_ERR("nrf_modem_lib_init failed, error: %d", err);
-		SEND_FATAL_ERROR();
-	}
+	net_mgmt_init_event_callback(&l4_cb, l4_event_handler, L4_EVENT_MASK);
+	net_mgmt_add_event_callback(&l4_cb);
+
+	net_mgmt_init_event_callback(&conn_cb, connectivity_event_handler, CONNECTIVITY_EVENT_MASK);
+	net_mgmt_add_event_callback(&conn_cb);
 
-	/* Setup a callback for the default PDP context. */
-	err = pdn_default_ctx_cb_reg(pdn_event_handler);
+	LOG_INF("Bringing network interface up");
 
+	err = net_if_up(net_if);
```

Comment:
Not related to this line directly but sample documentation says that [Network Interface](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/connectivity/networking/api/net_if.html#net-if-interface) is used by nrf7002dk setup only.
See https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/net/mqtt/doc/description.html

I guess example output would need an update too.

Reply:
Ill update the dependency bullett list and the log output. Ill wait until the wifi part has been added before I update the architecture diagram.

Reply:
Fixed